### ISSUE TITLE
Reorder causes folders to show up in a weird format on each device

### ISSUE THREAD

**Issue by srirambv** [brave/brave-ios#860]
### Description: 
There is a weird bug here when bookmarks/folders are reordered on one device
Device 1: iPhone 7+ sync chain creator with existing bookmarks
Device 2: iPad Pro created empty bookmark folder post sync
Reorder all bookmark folders to the end post sync causes the following

Device 1:
![image from ios](https://user-images.githubusercontent.com/17010094/52327005-86262180-2a10-11e9-9bda-9e26c0dfa7f9.png)
Device 2:
![file 5](https://user-images.githubusercontent.com/17010094/52327013-8de5c600-2a10-11e9-91ea-273dc745391f.jpg)

Closing the bookmark tray and reopening it shows correctly. Also some times it messes up the ordering and folders start showing up with weird colours

**Expected result:**
Order should be same on both devices

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Additional information**
https://github.com/brave/brave-ios/issues/772

**Comment by iccub** [comment]
Fixed via via https://github.com/brave/brave-ios/commit/09f57cb45a0aba93603ef5c0e3905b7ea2a492d2

**Comment by GeetaSarvadnya** [comment]
Verification passed on iPhone 8 with 1.9 (19.03.25.21). I have tried folders re-order between iPhone and Linux. @srirambv - In iPhone after bookmarks folder re-order color of folder changes is that expected?

iPhone:
![Bookmarks folders reorder](https://user-images.githubusercontent.com/38657976/55073128-f10be480-50b2-11e9-8db8-17d4cea28264.png)

Linux:
![image](https://user-images.githubusercontent.com/38657976/55072957-83f84f00-50b2-11e9-9ea6-0b83cdb394b2.png)



Verified passed with 1.9 (19.03.25.21) on iPad 5th Gen - reordering occurred as expected. I did encounter bookmark folder colors (noted below in https://github.com/brave/brave-ios/issues/860#issuecomment-477118376)

Verified passed on 1.9 (19.03.25.21) on iPhone SE

* Verification passed on iPhone 6/iPhone 7+ and iPad Pro all running 1.9(19.3.25.21)
